### DISCUSSION TITLE
Error: redefinition after use

### DISCUSSION THREAD

**Discussion by NTques** [OpenDDS/OpenDDS#4428] [Q&A] [answered]
Hello! I'm trying to compile ROS2 common interfaces files. Because for communicating between android device and ROS2 device.
But when I try to use arrays and sequences, many errors occured.

For example, when I declared typedefs about arrays and sequences, duplication error occured. Because several idl files use "typedef double double__36[36]".

So, I defined a typedef in one file and used it, because OpenDDS Java Binding does not support anonymous sequence.
But the error below occurred.

Error
```
line 6: redefinition after use, ::geometry_msgs::msg::dds_::double__36, ::geometry_msgs::msg::dds_, ::geometry_msgs::msg::dds_::double__36
```

Typedef IDL files.
```
// File Name: std_msgs_typedef.idl
module std_msgs {
  module msg {
    module dds_ {

      typedef sequence<octet> byteSeq;
      typedef sequence<float> float32Seq;
      typedef sequence<double> float64Seq;
      typedef sequence<int16> int16Seq;
      typedef sequence<int32> int32Seq;
      typedef sequence<int64> int64Seq;
      typedef sequence<int8> int8Seq;
      typedef sequence<uint16> uint16Seq;
      typedef sequence<uint32> uint32Seq;
      typedef sequence<uint64> uint64Seq;
      typedef sequence<uint8> uint8Seq;
    };
  };
};
```
```
// File Name: geometry_msgs_typedef.idl
module geometry_msgs {
  module msg {
    module dds_ {

      typedef double double__36[36];
    };
  };
};
```

And many .idl files include Typedef IDL files.

How can I solve this problem?

Sorry for my terrible English skills. Sorry if I use rude sentence...

**Comment by mitza-oci** [comment, marked as answer]
Is the same idl file being included multiple times?  Try adding the typical preprocessor conditional pattern that prevents multiple inclusion -- there are many of examples of this in OpenDDS.

```
#ifndef SYMBOL_NAME_UNIQUE_TO_THIS_FILE
#define SYMBOL_NAME_UNIQUE_TO_THIS_FILE

// existing contents of IDL file

#endif
```

**Reply by NTques** [reply]
Thank you so much! 
I'm not proficient in C/C++, so I didn't think about those aspects. 
Thanks to you, it was resolved well. 
Have a great day!
